src/cc/bcc_proc.c

```diff
@@ -148,39 +143,66 @@ int bcc_procutils_each_module(int pid, bcc_procutils_modulecb callback,
     name = buf;
     while (isspace(*name))
       name++;
+    mod.name = name;
     if (!bcc_mapping_is_file_backed(name))
       continue;
 
     if (strstr(name, "/memfd:")) {
-      char *memfd_name = _procutils_memfd_path(pid, inode);
+      char *memfd_name = _procutils_memfd_path(pid, mod.inode);
       if (memfd_name != NULL) {
         strcpy(buf, memfd_name);
         free(memfd_name);
-        name = buf;
+        mod.name = buf;
+        enter_ns = 0;
```

Comment:
The logic for enter_ns is not quite right here. If `enter_ns` here is set to 0, in the next iteration, it will remain 0, is this what you want. Also, for maps in a particular module, should be all files either enter_ns or not enter_ns?

Reply:
This is definitely not right - intention was to set `enter_ns` just for the iteration where memfd logic is triggered.

Before I dealt w/ rebase conflict due to the memfd logic here, `enter_ns` was true for all maps. When the memfd logic is triggered (`if (strstr(name, "/memfd:"))`), the name is replaced with something like `/proc/PID/fd/10`. I think we don't want to `enter_ns` here because `/proc/PID/fd/10` may not be same file as `/proc/PID/root/proc/PID/fd/10` due to pid namespace?

Reply:
Will modify to only set `enter_ns` to 0 for memfd iteration.